### PR TITLE
Fix workpackage not found when searching workpackage

### PR DESCRIPTION
**Pull request by SagarGi**
## Description

This PR fixes the the display of the text when there is search query typed already while searching the workpackes but still gives the wrong text as `Start typing to search`.

## Related Issue or Workpackage

- Fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/52956

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [x] Updated `CHANGELOG.md` file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
-- Replaced vue loading icon with default Nextcloud loading icon [#561](https://github.com/nextcloud/integration_openproject/pull/561)
-- Fixes: Signing terms and services for special user "OpenProject" when `terms_of_service` app is enabled [#552](https://github.com/nextcloud/integration_openproject/pull/552)
-- Fix error log when fetching non-existent work package [#573](https://github.com/nextcloud/integration_openproject/pull/573)
+- Fixes wrong option text while searching workpackage. [582](https://github.com/nextcloud/integration_openproject/pull/582)
```
Comment (individual-it) [inline CHANGELOG.md:8]:
```suggestion
- Fixes wrong option text while searching workpackage. [#582](https://github.com/nextcloud/integration_openproject/pull/582)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,9 +5,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
-- Replaced vue loading icon with default Nextcloud loading icon [#561](https://github.com/nextcloud/integration_openproject/pull/561)
-- Fixes: Signing terms and services for special user "OpenProject" when `terms_of_service` app is enabled [#552](https://github.com/nextcloud/integration_openproject/pull/552)
-- Fix error log when fetching non-existent work package [#573](https://github.com/nextcloud/integration_openproject/pull/573)
```
Comment (individual-it) [inline CHANGELOG.md:10]: why deleting those changelog lines?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (SagarGi) [reply CHANGELOG.md:10]: I have made a PR related to it: https://github.com/nextcloud/integration_openproject/pull/584

File: src/components/tab/SearchInput.vue [hunk 3]
```diff
@@ -130,6 +130,16 @@ export default {
...
+		getNoOptionText() {
+			if (this.noOptionTextState === NO_OPTION_TEXT_STATE.START_TYPING) {
+				return t('integration_openproject', 'Start typing to search')
+			} else if (this.noOptionTextState === NO_OPTION_TEXT_STATE.RESULT && this.searchResults.length === 0) {
+				return t('integration_openproject', 'There were no workpackages found')
+			} else {
+				// while workpackages are being searched we make the no text option empty
+				return t('integration_openproject', '')
+			}
+		},
```
Comment (nabim777) [inline src/components/tab/SearchInput.vue:142]: The condition looks long.. so can we use switch statement

File: src/components/tab/SearchInput.vue [hunk 4]
```diff
@@ -130,6 +130,19 @@ export default {
 			}
 			return ''
 		},
+		getNoOptionText() {
+			switch (this.noOptionTextState) {
+			case NO_OPTION_TEXT_STATE.START_TYPING :
+				return t('integration_openproject', 'Start typing to search')
+			case NO_OPTION_TEXT_STATE.RESULT :
+				if (this.searchResults.length === 0) {
+					return t('integration_openproject', 'There were no workpackages found')
```
Comment (nabim777) [inline src/components/tab/SearchInput.vue:139]:
```suggestion
					return t('integration_openproject', 'No workpackages found')
```

Comment (SagarGi) [reply src/components/tab/SearchInput.vue:139]: I think no need to change the text. I was approved in the meeting we are gonna keep this text. Better to check it manually.

Comment (nabim777) [reply src/components/tab/SearchInput.vue:139]: If so Ok then

File: src/components/tab/SearchInput.vue [hunk 5]
```diff
@@ -130,6 +130,19 @@ export default {
 			}
 			return ''
 		},
+		getNoOptionText() {
+			switch (this.noOptionTextState) {
+			case NO_OPTION_TEXT_STATE.START_TYPING :
+				return t('integration_openproject', 'Start typing to search')
```
Comment (nabim777) [inline src/components/tab/SearchInput.vue:136]:
```suggestion
				return t('integration_openproject', 'Start typing to search')
```

Screenshot from 2024-03-14 10-54-49

Two messages that seem to be saying the same thing

File: src/components/tab/SearchInput.vue [hunk 6]
```diff
@@ -130,6 +130,19 @@ export default {
...
+			case NO_OPTION_TEXT_STATE.RESULT :
+				if (this.searchResults.length === 0) {
+					return t('integration_openproject', 'There were no workpackages found')
+				}
+				break
+			}
+			// while workpackages are being searched we make the no text option empty
+			return t('integration_openproject', '')
```
Comment (nabim777) [inline src/components/tab/SearchInput.vue:144]:
```suggestion
				// while workpackages are being searched we make the no text option empty
			default: return ''
			}
```

Comment (nabim777) [inline src/components/tab/SearchInput.vue:144]:
```suggestion
			return  ''
```
I think no need to translate the empty string

**Review by nabim777 — APPROVED** [review]

**Pull request merged**